FWCore/Framework/src/global/OutputModuleBase.cc

```diff
@@ -157,6 +158,13 @@ namespace edm {
       ProductSelector::checkForDuplicateKeptBranch(desc, trueBranchIDToKeptBranchDesc);
 
       EDGetToken token;
+
+      std::vector<std::string> missingDictionaries;
+      if (!checkDictionary(missingDictionaries, desc.className(), desc.unwrappedType())) {
+        std::string context("Calling OutputModuleBase::keepThisBranch, checking dictionaries for kept types");
+        throwMissingDictionariesException(missingDictionaries, context);
+      }
```

Comment:
This is not related to ProcessBlock. I had to modify some other unrelated lines of code in this same OutputModuleBase function that occur later. I noticed that for "one" OutputModules we make this check, but for "global" and "limited" we didn't. It seemed like they ought to be consistent. Possibly I am wrong. Is there a reason for them to be different? It makes sense for PoolOutputModule to check this for products it will need to write. PoolOutputModule is a "one". At this point, all global or limited output modules are in Framework tests except for AsciiOutputModule. Let me know if you want me remove this check for global and limited.

Reply:
I was mostly curious, your explanation makes sense (thanks!). @Dr15Jones, do you have anything to add?

Reply:
Being consistent is good.